src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/BlockMerkleRootRule.cs

```diff
@@ -7,30 +7,29 @@
 
 namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
 {
+    /// <summary>
+    /// Check for merkle tree malleability (CVE-2012-2459): repeating sequences
+    /// of transactions in a block without affecting the merkle root of a block,
+    /// while still invalidating it.
+    /// </summary>
```

Comment:
This rule does more than that, the primary intention of this is to check if the merkle root  is actually correctly calculated and only then it checks for that vulnerability.